common/tasks/interleaved_weighted_round_robin_test.go

```diff
@@ -395,6 +403,132 @@ func (s *interleavedWeightedRoundRobinSchedulerSuite) TestUpdateWeight() {
+	s.scheduler.Submit(mockTask3)
+	taskWG.Wait()
+
+	var channelWeights []int
+	for _, channel := range s.scheduler.channels() {
+		channelWeights = append(channelWeights, channel.Weight())
+	}
+	s.Equal([]int{5, 5, 5, 3, 5, 3, 2, 5, 3, 2, 1}, channelWeights)
+
+	s.ts.Advance(30 * time.Minute)
+
+	// Only add tasks to first two channels.
+	taskWG.Add(2)
+	s.scheduler.Submit(mockTask0)
+	s.scheduler.Submit(mockTask1)
+	taskWG.Wait()
+
+	// Advance time past 1 hour. This will make other two channels inactive for more than 1 hour.
+	s.ts.Advance(31 * time.Minute)
+
+	s.Eventually(func() bool {
+		channelWeights = []int{}
+		for _, channel := range s.scheduler.channels() {
+			channelWeights = append(channelWeights, channel.Weight())
+		}
+		if !slices.Equal([]int{5, 5, 5, 3, 5, 3, 5, 3}, channelWeights) {
+			return false
+		}
+		return true
+	}, 30*time.Second, 100*time.Millisecond)
+
+	// set the number of pending task back
+	atomic.AddInt64(&s.scheduler.numInflightTask, -1)
+}
+
+func (s *interleavedWeightedRoundRobinSchedulerSuite) TestInactiveChannelDeletionDelayNotProvided() {
+	s.scheduler = NewInterleavedWeightedRoundRobinScheduler(
+		InterleavedWeightedRoundRobinSchedulerOptions[*testTask, int]{
+			TaskChannelKeyFn:      func(task *testTask) int { return task.channelKey },
+			ChannelWeightFn:       func(key int) int { return s.channelKeyToWeight[key] },
+			ChannelWeightUpdateCh: s.channelWeightUpdateCh,
+			// Not setting InactiveChannelDeletionDelay
+		},
+		Scheduler[*testTask](s.mockFIFOScheduler),
+		log.NewTestLogger(),
+	)
+	s.scheduler.ts = s.ts
+	s.mockFIFOScheduler.EXPECT().Start()
+	s.scheduler.Start()
+	s.mockFIFOScheduler.EXPECT().Stop()
+
+	var taskWG sync.WaitGroup
+	s.mockFIFOScheduler.EXPECT().Submit(gomock.Any()).Do(func(task Task) {
+		taskWG.Done()
+	}).AnyTimes()
+
+	// need to manually set the number of pending task to 1
+	// so schedule by task priority logic will execute
+	atomic.AddInt64(&s.scheduler.numInflightTask, 1)
+
+	mockTask0 := newTestTask(s.controller, 0)
+	mockTask1 := newTestTask(s.controller, 1)
+	mockTask2 := newTestTask(s.controller, 2)
+	mockTask3 := newTestTask(s.controller, 3)
+
+	taskWG.Add(4)
+	s.scheduler.Submit(mockTask0)
+	s.scheduler.Submit(mockTask1)
+	s.scheduler.Submit(mockTask2)
+	s.scheduler.Submit(mockTask3)
+	taskWG.Wait()
+
+	var channelWeights []int
+	for _, channel := range s.scheduler.channels() {
+		channelWeights = append(channelWeights, channel.Weight())
+	}
+	s.Equal([]int{5, 5, 5, 3, 5, 3, 2, 5, 3, 2, 1}, channelWeights)
+
+	s.ts.Advance(30 * time.Minute)
+
+	// Only add tasks to first two channels.
+	taskWG.Add(2)
+	s.scheduler.Submit(mockTask0)
+	s.scheduler.Submit(mockTask1)
+	taskWG.Wait()
+
+	// Advance time past 1 hour. This will make other two channels inactive for more than 1 hour.
+	// But this time, those channels should not be deleted as we haven't provided InactiveChannelDeletionDelay
+	s.ts.Advance(31 * time.Minute)
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
what  is this sleep for?

Reply:
This is to ensure the cleanup thread has not woken up and cleaned up this channel.